src/hera/workflows/artifact.py

```diff
@@ -112,11 +118,21 @@ def _build_artifact_paths(self) -> _ModelArtifactPaths:
         return _ModelArtifactPaths(**artifact.dict())
 
     def as_name(self, name: str) -> _ModelArtifact:
-        """Returns a copy of the current artifact but named based on the specified `name`."""
+        """DEPRECATED, use with_name.
+
+        Returns a 'built' copy of the current artifact, renamed using the specified `name`.
+        """
+        logger.warning("'as_name' is deprecated, use 'with_name'")
         artifact = self._build_artifact()
         artifact.name = name
         return artifact
 
+    def with_name(self, name: str) -> Artifact:
+        """Returns a copy of the current artifact, renamed using the specified `name`."""
+        artifact = self.copy(deep=True)
+        artifact.name = name
+        return artifact
```

Comment:
Small request - can we add a `with_path` as well? Heard some feedback from the community that it would be very helpful, it was part of v4

Reply:
The path is dictated by the template though, that only worked in v4 because the template/tasks were linked together